### ISSUE TITLE
Support round corner for 2d rectangle shape

### ISSUE THREAD

**Issue by lewiszlw** [bevyengine/bevy#7972]
## What problem does this solve or what need does it fill?

as title.

**Comment by lewiszlw** [comment]
> Previously attempted in #3991: good discussion there.

Thanks, seems it's only about ui, I'd like round corner for `Quad` shape.

**Comment by alice-i-cecile** [comment]
Yep, seems reasonable (although probably make a different shape?). Still useful insight there IMO.

**Comment by mockersf** [comment]
I agree on making it a new shape rather than making `Quad` more complex
